src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -292,6 +300,15 @@ public ProgrammingExercise importProgrammingExercise(ProgrammingExercise origina
         newProgrammingExercise = programmingExerciseImportBasicService.importProgrammingExerciseBasis(originalProgrammingExercise, newProgrammingExercise);
         importRepositories(originalProgrammingExercise, newProgrammingExercise);
 
+        if (setTestCaseVisibilityToAfterDueDate) {
+            Set<ProgrammingExerciseTestCase> testCases = this.programmingExerciseTestCaseRepository.findByExerciseId(newProgrammingExercise.getId());
+            for (ProgrammingExerciseTestCase testCase : testCases) {
+                testCase.setVisibility(Visibility.AFTER_DUE_DATE);
+            }
+            List<ProgrammingExerciseTestCase> updatedTestCases = programmingExerciseTestCaseRepository.saveAll(testCases);
+            newProgrammingExercise.setTestCases(Set.copyOf(updatedTestCases));
```

Comment:
```suggestion
            newProgrammingExercise.setTestCases(new HashSet<>(updatedTestCases));
```
Set.copyOf is unmodifiable, which could lead to problems in the future if we want to change test cases after this method